background.ts

```diff
@@ -238,6 +242,52 @@ async function checkBackendValid() {
   }
 }
 
+/**
+ * Removes the custom protocol prefix (app://) from the provided url
+ * @param url The requested URL
+ * @returns A path with the custom protocol (app://) removed
+ */
+function stripCustomProtocol(url: string) {
+  const protocol = 'app://';
+
+  if (!url?.startsWith(protocol)) {
+    return url;
+  }
+
+  return url.replace(protocol, '');
+}
+
+/**
+ * Create a URL that consists of a base combined with the provided path
+ * @param path The destination path for the requested resource
+ * @returns A URL that consists of the combined base (http://localhost:8888)
+ * and provided path
+ */
+function redirectedUrl(path: string) {
+  const base = 'http://localhost:8888/';
+
+  return `${ base }${ path }`;
+}
+
+/**
+ * Set up a protocol handler app:// when running Rancher Desktop in a dev
+ * environment
+ */
+function setupDevProtocolHandler() {
+  Electron.protocol.registerHttpProtocol('app', (request, callback) => {
+    const path = stripCustomProtocol(request.url).replaceAll('index.html/', '');
```

Comment:
Huh, do we normally get `app://index.html/index.js` or something?

Reply:
That's correct, our requests are structured like this `app://index.html/_nuxt/commons/app.js`. I thought it would be simplest to remove `index.html/` since a request to `http://localhost:8888` defaults to index anyway..